moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -82,39 +82,18 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
     : ActionBasedControllerHandleBase(name, logger_name), node_(node), done_(true), namespace_(ns)
   {
     controller_action_client_ = rclcpp_action::create_client<T>(node_, getActionName());
+    last_exec_ = moveit_controller_manager::ExecutionStatus::SUCCEEDED;
+  }
 
-    unsigned int attempts = 0;
-    double timeout;
-    node_->get_parameter_or("trajectory_execution.controller_connection_timeout", timeout, 15.0);
-
-    if (timeout == 0.0)
-    {
-      while (rclcpp::ok() && !controller_action_client_->wait_for_action_server(std::chrono::seconds(5)))
-      {
-        RCLCPP_WARN_STREAM(LOGGER, "Waiting for " << getActionName() << " to come up");
-      }
-    }
-    else
-    {
-      while (rclcpp::ok() &&
-             !controller_action_client_->wait_for_action_server(std::chrono::duration<double>(timeout / 3)) &&
-             ++attempts < 3)
-      {
-        RCLCPP_WARN_STREAM(LOGGER, "Waiting for " << getActionName() << " to come up");
-      }
-    }
+  bool isConnected() const
```

Comment:
I'm sorry if this is obvious, but what is the user of this?  As we are no longer waiting for the connection in the constructor does the user need to call this before calling other methods on this?  Should each method that interacts with this check this before doing whatever it is going to do?

Would you mind adding some documentation (even just a comment) about the pre-conditions of each method here?

Reply:
After looking at how this function is used some more, I think this is a good thing to change here.

Right now `isConnected` is only used by the MoveItSimpleControllerManager, and it isn't part of the interface of the [MoveItControllerHandle base class](https://github.com/ros-planning/moveit2/blob/54c6e08ead8d5888198cb819cd933cb6229a3077/moveit_core/controller_manager/include/moveit/controller_manager/controller_manager.h#L105).

The expectation established in the base class is that `sendTrajectory` returns `false` if the controller cannot accept the trajectory, so it should not be a requirement to do a separate check before that to determine if the controller handle is ready to execute trajectories.

In the current ActionBasedControllerHandle implementation the class constructor leaves class members not fully initialized if the action server wasn't found within some timeout, which leaves it in sort of a limbo state. There isn't a way to have the handle retry connecting later if needed (other than knowing that you might have to create a new instance of the handle later). This doesn't seem like a good way to handle this.

I think the right way forward is to delete `isConnected` and modify MoveIt SimpleControllerManager to have it always create the controller handles regardless of the availability of the associated action servers.

Reply:
^I'm good with what you wrote, Joe, as long as `sendTrajectory` prints a nice error if the action server isn't running

Reply:
^What about you, @JafarAbdi ?

Reply:
> as long as sendTrajectory prints a nice error if the action server isn't running

Since each controller handle has to provide its own implementation of `sendTrajectory`, it's the handle's responsibility to log an error if trajectory execution failed due to a missing action server. The existing handles in the `moveit_simple_controller_manager` package already do this.